kratos/python_scripts/multistage_analysis.py

```diff
@@ -0,0 +1,149 @@
+        '''Creates and returns the execution list.
+        This method creates the execution list, either from a user-defined execution list or,
+        if this is not provided, from the stages declaration order.
+        '''
+
+        # Check if the execution list has been already created
+        if not hasattr(self, "__execution_list"):
+            # Default case in which the execution list is provided by the user
+            if self.settings.Has("execution_list"):
+                self.__execution_list = self.settings["execution_list"].GetStringArray()
+            # If not provided, create an auxiliary execution list from the stages declaration order
+            else:
+                KratosMultiphysics.Logger.PrintInfo("'execution_list' is not provided. Stages will be executed according to their declaration order.")
+                self.__execution_list = []
+                for stage_name in self.settings["stages"].keys():
+                    self.__execution_list.append(stage_name)
+
+        return self.__execution_list
+
+    def __GetModelers(self):
+        '''This method creates the modelers at the preprocess execution point.'''
+
+        execution_point_settings = self.settings["stages"][self.GetCurrentStageName()]["stage_preprocess"]
+
+        factory = KratosModelParametersFactory(self.model)
+        return factory.ConstructListOfItems(execution_point_settings["modelers"])
+
+    def __GetOperations(self, execution_point):
+        '''This method creates the operations at any execution point.'''
+
+        if execution_point not in ["stage_preprocess","stage_postprocess"]:
+            err_msg = f"Wrong execution point '{execution_point}'. Supported ones are 'stage_preprocess' and 'stage_postprocess'."
+            raise Exception(err_msg)
+        execution_point_settings = self.settings["stages"][self.GetCurrentStageName()][execution_point]
+
+        factory = KratosModelParametersFactory(self.model)
+        return factory.ConstructListOfItems(execution_point_settings["operations"])
+
+    def __CreateStagesMap(self):
+        '''This method creates the stages map.
+        The keys of the map are the stage names, which are retrieved from the execution list (see __GetExecutionList()).
+        The values of the map are the stages instances, which are created by importing its corresponding analysis stage Python module.
+        '''
+
+        stages_map = {}
+        for stage_name in self.__GetExecutionList():
+            # Get the Python module name and class name implementing each analysis stage
+            # Note that we assume that the class name is the provided module (file) name in CamelCase
+            analysis_stage_module_name = self.settings["stages"][stage_name]["analysis_stage"].GetString()
+            analysis_stage_class_name = analysis_stage_module_name.split('.')[-1]
+            analysis_stage_class_name = ''.join(x.title() for x in analysis_stage_class_name.split('_'))
```

Comment:
So you mean having a `CreateMethod` in our current `MainKratos.py` files. Doesn't sound bad to me as these were unified thanks to the  `analysis_stage` field. However, I wouldn't make it mandatory just in case someone is currently using custom `MainKratos.py` files. 

Hence, we can do both options, first we check for a `Create` in the corresponding `analysis_stage` Python module. If not found, we try to create it like we are doing in here. Makes sense for you?

Reply:
@sunethwarna I added the `Create` option. However, we do it in the other way around, first we check if the "official" naming convention is followed. If not, we try to use the `Create`. If none of these work an error is thrown.